frontend/packages/ux-editor/src/components/Elements/Elements.tsx

```diff
@@ -1,24 +1,65 @@
 import React from 'react';
 import { ConfPageToolbar } from './ConfPageToolbar';
 import { DefaultToolbar } from './DefaultToolbar';
-import { Heading, Paragraph } from '@digdir/design-system-react';
-import { useText, useAppContext } from '../../hooks';
+import { Alert, Heading, Paragraph } from '@digdir/design-system-react';
+import { useAppContext } from '../../hooks';
 import { LayoutSetsContainer } from './LayoutSetsContainer';
 
 import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
 import classes from './Elements.module.css';
 
 import { useCustomReceiptLayoutSetName } from 'app-shared/hooks/useCustomReceiptLayoutSetName';
+import { useProcessTaskTypeQuery } from '../../hooks/queries/useProcessTaskTypeQuery';
+import { StudioSpinner } from '@studio/components';
+import { useTranslation } from 'react-i18next';
 
-export const Elements = () => {
+export const Elements = (): React.ReactElement => {
+  const { t } = useTranslation();
   const { org, app } = useStudioEnvironmentParams();
   const { selectedFormLayoutSetName, selectedFormLayoutName } = useAppContext();
-  const existingCustomReceiptName: string | undefined = useCustomReceiptLayoutSetName(org, app);
 
+  const {
+    data: processTaskType,
+    isPending: isFetchingProcessTaskType,
+    isError: hasProcessTaskTypeError,
+  } = useProcessTaskTypeQuery(org, app, selectedFormLayoutSetName);
+
+  const existingCustomReceiptName: string | undefined = useCustomReceiptLayoutSetName(org, app);
   const hideComponents =
     selectedFormLayoutName === 'default' || selectedFormLayoutName === undefined;
 
-  const t = useText();
+  if (isFetchingProcessTaskType) {
+    return (
+      <div className={classes.root}>
+        <StudioSpinner
+          spinnerTitle={t('schema_editor.loading_available_components')}
+          showSpinnerTitle
+        />
+      </div>
+    );
+  }
+
+  if (hasProcessTaskTypeError) {
+    return (
+      <div>
+        <LayoutSetsContainer />
```

Comment:
Nice! The second reason I did this was because the code below does not execute unless the `processTaskType` has a value. Then we do not compare `processTaskType (undefined) === 'payment'`. 😊